contrib/cf-remote/cf_remote/main.py

```diff
@@ -18,23 +21,60 @@ def get_args():
     ap.add_argument(
+        "--version", "-V", help="Print or specify version", nargs="?", type=str, const=True)
+
+    command_help_hint = "Commands (use %s COMMAND --help to get more info)" % os.path.basename(sys.argv[0])
+    subp = ap.add_subparsers(dest="command",
+                             title=command_help_hint)
+
+    sp = subp.add_parser("info", help="Get info about the given hosts")
+    sp.add_argument("--hosts", "-H", help="Which hosts to get info for", type=str, required=True)
+
+    sp = subp.add_parser("install", help="Install CFEngine on the given hosts")
+    sp.add_argument("--edition", "-E", help="Enterprise or community packages", type=str)
+    sp.add_argument("--package", help="Local path to package for transfer and install", type=str)
+    sp.add_argument("--hub-package", help="Local path to package for --hub", type=str)
+    sp.add_argument("--client-package", help="Local path to package for --clients", type=str)
+    sp.add_argument("--bootstrap", "-B", help="cf-agent --bootstrap argument", type=str)
+    sp.add_argument("--clients", "-c", help="Where to install client package", type=str)
+    sp.add_argument("--hub", help="Where to install hub package", type=str)
+    sp.add_argument(
         "--demo", help="Use defaults to make demos smoother (NOT secure)", action='store_true')
-    ap.add_argument(
+    sp.add_argument(
         "--call-collect", help="Enable call collect in --demo def.json", action='store_true')
-    ap.add_argument(
-        "--version", "-V", help="Print or specify version", nargs="?", type=str, const=True)
-    ap.add_argument("command", help="Action to perform (info|install|packages|run|sudo|scp)", type=str, nargs='?')
-    ap.add_argument("args", help="Arguments", type=str, nargs='*')
+
+    sp = subp.add_parser("packages", help="Get info about available packages")
+    sp.add_argument("--edition", "-E", help="Enterprise or community packages", type=str)
+    sp.add_argument("tags", metavar="TAG", nargs="*")
+
+    sp = subp.add_parser("run", help="Run the command given as arguments on the given hosts")
+    sp.add_argument("--hosts", "-H", help="Which hosts to run the command on", type=str, required=True)
+    sp.add_argument("args", help="Arguments", type=str, nargs='*')
+
+    sp = subp.add_parser("sudo",
+                         help="Run the command given as arguments on the given hosts with 'sudo'")
+    sp.add_argument("--hosts", "-H", help="Which hosts to run the command on", type=str, required=True)
+    sp.add_argument("args", help="Arguments", type=str, nargs='*')
+
+    sp = subp.add_parser("scp", help="Copy the given file to the given hosts")
+    sp.add_argument("--hosts", "-H", help="Which hosts to copy the file to", type=str, required=True)
+    sp.add_argument("args", help="Arguments", type=str, nargs='*')
+
+    sp = subp.add_parser("spawn", help="Spawn hosts in the clouds")
+    sp.add_argument("--list-platforms", help="List supported platforms", action='store_true')
```

Comment:
The python is beyond my ability to provide a meaningful review, but i see here spawn hosts in the clouds, and the data above had aws_platforms. Is there a pre-thought (but not yet implemented) path for different clouds like Azure, digitalocean or _____?

Reply:
Yes, exactly. And *Vagrant*.